index.js

```diff
@@ -751,6 +795,18 @@ function parse (args, opts) {
     return isSet
   }
 
+  function hasAnyFlag (key) {
+    var isSet = false
+    // XXX Switch to [].concat(...Object.values(flags)) once node.js 6 is dropped
```

Comment:
I like the method name `hasUnknownOption`, which would take an `arg` and `arg + 1` and:

1. if argument is form `-foo`, and `arg + 1` starts with `-`, it would ensure that `f`, `o`, and `o`, are known keys.
2. if argument is form `--foo`, or `--foo=bar`, it would check for `foo`.

Rather than using the approach:

```
var toCheck = [].concat(...Object.keys(flags).map(k => flags[k]))
```

I think you'd be best to check `opts.key`, which will be set even if you only call `yargs.option('foo', {describe: 'foo key'})`.

☝️ I think if you use this approach you can just call:

```
checkAllAliases(key, opts.key)
```